woocommerce/woocommerce-ios
# [IPPInAppFeedback] Orders screen banner placement

Closes: #8636 
Part of: https://github.com/woocommerce/woocommerce-ios/issues/8530

## Description

This PR adds the in-app feedback banner to the Order view, as well as the buttons to launch a survey and dismiss the banner.

At the moment the banner is not dismissible. Dismiss will be part of a [different PR](https://github.com/woocommerce/woocommerce-ios/issues/8589).

## Changes
- Adds a `.IPP` case as new `FeedbackType`
- Upon configuring `OrderListViewController`, we set the existing top banner view (generally used to display the Order Creation banner, or an error) to a new IPP banner survey. This uses the existing `TopBannerView` and `TopBannerViewModel` components.
- We have a new Publisher `hideIPPFeedbackBanner`, so we can listen to changes and set priorities on the different banners that should display in the view.
- The banner has an `ActionButton` that will launch the feedback survey via the existing `SurveyCoordinatingController` component.

## Testing instructions

1. Go to Menu > Orders > See the banner at the top of the screen
2. Tap "Share Feedback", a WebView showing a survey form will shop up. Scroll down and tap "Finish Survey", a "Feedback Sent" view will be shown and you'll be redirected to the Orders page.

## Screenshots
![Simulator Screen Recording - iPhone 11 Pro - 2023-01-16 at 12 40 23](https://user-images.githubusercontent.com/3812076/212600007-00e6d0de-bff9-4cf7-8b0e-2a52aa45ed1d.gif)


---
